sw/host/opentitanlib/src/crypto/spx.rs

```diff
@@ -86,6 +94,13 @@ impl SpxKeypair {
         SpxSignature(sphincsplus::spx_sign(&self.sk, full_message).unwrap())
     }
 
+    /// Sign `message` using the secret key in "prehash" mode with an empty context.
+    pub fn sign_prehash(&self, oid: &[u8], message: &[u8]) -> SpxSignature {
```

Comment:
How do you want to connect this up to the CLI?

Obviously, we could have a command-line switch that selects among raw, pure and pre-hash.  We could also attach some meta-data to the key to indicate its usage constraint.

Reply:
Yes, I think a mode option with values raw, pure, and pre-hash, defaulting to pure, would make sense.